### ISSUE TITLE
Switching Between Shipping Methods Don't Get Synced To Existing Populated Carts.

### ISSUE THREAD

**Issue by Jplus2** [woothemes/woocommerce#7101]
Example switching between free shipping and flat rate shipping

*for Flat Rate make sure that at least one class is present or "Any Class" has cost and handing fee, this test is done with the use of "Any Class"

Given the customer already has populated products on his/her shopping cart, and he has not yet made a previous order (meaning not entered a billing address or shipping address yet)

* Customer A has product A on his cart and previously it was set to free shipping
* Then on the back end the admin sets shipping from free shipping to flat rate shipping
* Customer refreshes the page, removes the products and re add them, admin clicked clear the transients, admin clicked clear expired transients, admin clicked clear all sessions*, shipping debug mode is activated (checked), the cart remains on free shipping, also on the checkout page*
* admin updates product A on the backend (click the update button) then customer refreshes the cart and flat rate shipping is applied
* if admin did not update product A on the backend (meaning it says free shipping instead of flat rate) after the order is successful, the checkout thank you page correctly uses the flat rate shipping (as oppose to the cart and checkout page displaying free shipping)

Now when the customer has previously set billing address, the behavior changes, when you clicked clear all sessions button on the tools tab on the system status, the cart still displays free shipping but if you go to checkout page, it does use flat rate shipping, then if you go back to cart it correctly uses flat rate shipping

Also when billing address is previously entered, clicking clear all sessions button alone does the trick, but sometimes it works, sometimes it don't (I can't find a pattern as to when it works and when it doesn't)

Also if more than 1 product is added to the cart, as opposed to only 1 product, the behavior changes as well, sometimes clicking, clear transients, clear expired transients and clear all sessions button doesn't work, sometimes it does.

Bottom line, the only predictable  way of making sure the existing carts applies the changed shipping method is by updating one of the products inside a cart (for multiple products at least one needs to be updated)

Other than that, making the cart synced with the changed shipping method is very unpredictable.

Wordpress: 4.1
WooCommerce: 2.2.10
No caching plugin installed

**Comment by mikejolley** [comment]
I tried your example, and turned off free shipping and turned on flat rate; it updated immediately.

> Then on the back end the admin sets shipping from free shipping to flat rate shipping

It's not clear from your example, but if you are talking about default shipping method it shouldn't change the customers selection if they have already made a choice. The only time it *should* re-calc and re-set default is if the cart contents change.

So for your example,

- If I'm in my cart and I've chosen free shipping already, your changes won't affect me
- If I'm in my cart and I've chosen free shipping, then you change default, and I change contents, my method will update

Also note, clear all sessions for logged out Customer A means his cart is destroyed. Also confirmed this for logged in users.

**Comment by Jplus2** [comment]
Apologies if I am not so clear. Below is my step by step scenario

* I have 2 browsers opened, chrome for the admin, ff for the customer
* Originally the shipping method is set to free shipping, that is, disabled flat rate shipping by unchecking the "Enable this shipping method" on the flat rate tab, enabling the free shipping by checking the same checkbox on the free shipping tab, then setting the default to free shipping on the shipping options tab
* customer (ff) adds products to cart, which applies free shipping only
* admin (chrome) sets shipping from free shipping to flat rate shipping, that is disabling free shipping, then enabling flat rate shipping, then setting the default to flat rate shipping
* customer (ff) still sees free shipping on his/her cart

Try changing shipping method consecutively, that is, change from free to flat rate, then flat rate to free, etc.. then you will see what I mean

Hope this clear things out.
